### DISCUSSION TITLE
Wifi setup

### DISCUSSION THREAD

**Discussion by vo5tr0** [PricelessToolkit/SOILSENS-V5W#6] [General]
### Re-entering Configuration Mode
After accidentally entering an incorrect MAC address (BSSID) during setup, I struggled to get the sensor back into configuration mode. Here’s the solution:

1. Unplug the battery.
2. Press and hold the calibration button.
3. While holding the button, plug the battery back in.
4. Release the button after 5 seconds.

_(Note: PricelessToolkit has already updated their README to include this information. Kudos to them for the quick update!)_

### The Wi-Fi Connection Issue
The main issue I faced was a failed Wi-Fi connection due to an incorrect MAC address (BSSID). As mentioned in the README, you need to provide the MAC address of your Wi-Fi access point.

I use a Mesh Wi-Fi system from AVM, and the only access point in range was a mesh repeater. Naturally, I assumed I should use the MAC address of the repeater. However, it turns out that even though my main router (Mesh Master) was out of range, I still needed to use its MAC address for the sensor to connect successfully.

I hope this helps anyone facing similar problems.

**Comment by PricelessToolkit** [comment]
Thank you for sharing!

**Comment by mafsgraz** [comment]
thanks for sharing Julian! 
i already needed these instructions several times now, but unfortunately i could not connect my sensor to my wifi network until now,
after saving my configuration the green led keeps shining and the sensor does not connect.

i'm not sure which mac address is the correct one, so let me describe my setup and i hope somebody can give me advice:

i am using a 5G Modem to establish internet connection that is also working as dhcp server for my network. i have reserved a range of IP addresses for some of my smart home devices that get assigned static IP's. 

for wifi, i am using a unifi UAP AC Pro access point, also with static IP. (no mesh, just this one access point)

i tried several variants without success and am hoping for advice - here is my current configuration:

Local IP: <static IP i chose for the sensor>

Gateway: IP address of the 5G modem 

MAC address: mac address of the unifi access point

Wifi Channel: channel of the 2.4 Ghz Network (the Gateway operates 2 networks in 2.4 and 5 GHz.

did anyone have similar issues and has an idea what the correct configuration should be?

**Reply by PricelessToolkit** [reply]
Hi, when the green LED stays on, it can mean:  
1. Out of range.  
2. Wi-Fi or MQTT credentials issue, so it can't connect to Wi-Fi or the MQTT server.  

If you briefly push the RST button, does the green LED continue to stay on?

**Reply by mafsgraz** [reply]
hi, my guess are a misconfiguration of the credentials - i am testing 3m away from the access point so that shouldn't be a problem.

pushing the RST button doesn't make a change on the green LED.

**Reply by PricelessToolkit** [reply]
Yep
Gateway of your network, channel, and MAC of your AC Pro, and one more important thing: the sensor name needs to be without spaces, or Home Assistant will not autodiscover it.

**Reply by mafsgraz** [reply]
thank you for the feedback! unfortunately, i had no success so far. :-( 
even tried configuring from different browsers and OS's, but the result is always the same.

thank you for your support, will try again next week.

**Comment by Mr-Basie** [comment]
I have trouble connecting to wifi.

Do the sensor have trouble with ssid with "å,æ,ø and a space"?

And what is the Local ip supposed to be?

Thanks! cool product anyway.

**Reply by PricelessToolkit** [reply]
That's right, no spaces! I can’t speak for other characters, though. For local IPs like the ones your phone or PC gets when connecting to Wi-Fi via DHCP (e.g., '192.168.1.50-192.168.1.254') you’ll need to assign an IP address outside the DHCP range. For example, you could use '192.168.1.49'. You can check your Wi-Fi router to find the DHCP range.  Thanks

**Comment by Focusinger** [comment]
I can't get them to work. Maybe you can told exactly which IP Adress I have to enter and which Mac Adress and which MQTT Adress. I tried everything but nothing worked. The YouTube Video to set them up is not very helpful, also the explanation of the Web Configuration for Wifi Mode on GitHub.

**Reply by PricelessToolkit** [reply]
SSID has space in it "it is not a good practice", sensor firmware can't handle it. Can you change your SSID ? replace space with "_" or "-"

**Reply by Focusinger** [reply]
Then I had to rebind all my Network Stuff. I never had a Problem with that.

**Reply by PricelessToolkit** [reply]
That is not the correct IP address. The IP address shown in your screenshot is the one assigned by the sensor to your phone. You need to assign an IP address outside the DHCP range but within the same subnet as the one your router assigns to your phone or computer. When you connect to your home router, what IP address do you get?

**Reply by Focusinger** [reply]
I don't understand what IP is required. The Router IP, my Phone IP? which IP do you mean? Where can i see it? I Connect to my Router with ssid and password. i don't get an IP Adress.

**Reply by PricelessToolkit** [reply]
When your router's DHCP assigns IP addresses to connected devices, it usually assigns them within the range of 192.168.178.X up to 192.168.178.254. Think of the subnet as the last part of the IP address, which can range from 1 to 254. Normally, DHCP servers do not assign IPs starting from 1 or up to 254, so you usually have free IP addresses that won't be automatically assigned to any connected device. You need to choose an IP from this free range. You can check the DHCP range in your access point settings.

So:

IP: 192.168.178.X (a free one that is not assigned to any other device and will not be!)
Gateway: 192.168.178.1
Subnet: 255.255.255.0

**Reply by Focusinger** [reply]
Okay, i tried it out tomorrow. thanks for your help.

**Reply by Focusinger** [reply]
I can't get them to work. I renamed my ssid without spaces. Which Mac Adress do I use? This one?
![Screenshot_20241227-103654](https://github.com/user-attachments/assets/949d7292-5b82-4f54-a5e0-18e32995d620)

or this one:
![Screenshot_20241227-103833](https://github.com/user-attachments/assets/d3edae2f-6334-4b12-833a-46ce6362901c)

And which MQTT Server? This one?
![Screenshot_20241227-103912](https://github.com/user-attachments/assets/c3a14270-6a70-43e0-917d-bdcce106a4ea)

or this one?
![Screenshot_20241227-104000](https://github.com/user-attachments/assets/6330a744-8033-4146-9ef3-4e696550e092)

And is there any Change after correct Settings in the LED on the Sensor? It only lights green after Configuration Saved. Should it blink or go off?

**Comment by PricelessToolkit** [comment]
2.4Ghz Mac .......58:39
If you installed the MQTT server (Mosquitto broker) in Home Assistant, use the IP address of your Home Assistant as MQTT SERVER.
PS: I made a mistake in the manual regarding SSIDs. You can have spaces in the SSID, but you cannot have spaces in the sensor name. I will update the manual accordingly Sorry.

**Reply by Focusinger** [reply]
It works now, but I have much Trouble with all my Network Devices, Cameras, etc. because i renamed my ssid without spaces. 😞

**Reply by PricelessToolkit** [reply]
Sorry for that!! You can have spaces in the SSID, but you cannot have spaces in the sensor name because it uses a sensor name for MQTT topic too.

**Comment by muradsam5555** [comment]
[
![IMG_9396](https://github.com/user-attachments/assets/17401b68-7c15-4fe6-9cce-418ea8e28c8a)
](url)

Please help after entering all the data the wi-fi mode on my Serial Port gives this message in cycles

**Reply by PricelessToolkit** [reply]
when programming with Arduino check "erase memory before programming"

**Reply by PricelessToolkit** [reply]
plus in the log, you can see "invalid IP Configuration"

**Comment by norfolkmustard** [comment]
Having trouble getting this working also unfortunately. The config requirements seem overly complex - why can't it be auto-assigned an IP address by DHCP and get subnet/gateway etc that way? If the wifi network is only 2.4GHz, does it still need the wifi MAC ?

If the green light stays on, is there a way to pinpoint why - which config field needs changing?

**Reply by PricelessToolkit** [reply]
1. If you are on Windows you can check the IP subnet, by taping in the terminal "ifconfig /all"
or If you don't have 255 wifi connected devices you can use "192.168.x.254"

2. I think you know the HA IP address and MQTT credentials
3. For Mac address and channel use "wifianalyzer" app "For android" if you don't have access to the access point menu.

Mac address is essential for fast connect.

probably you are inserting MAC and channel incorrectly.

**Comment by muradsam5555** [comment]
![IMG_0023](https://github.com/user-attachments/assets/b782ccfc-9e2c-4917-9c59-8439ea4b1280)
	1.	Why am I getting this error, and how can I fix it?
	2.	Could someone provide a detailed step-by-step guide on what exactly to enter in the web interface when the ESP32 goes into Configuration Mode and I press Wi-Fi?

**Reply by PricelessToolkit** [reply]
1.  "broker.hivemq.com"  is the address of your MQTT server?
2. https://github.com/PricelessToolkit/SOILSENS-V5W?tab=readme-ov-file#1-configuration

![hotspot2](https://github.com/user-attachments/assets/543d4412-452a-4226-bb5e-57db3ab5030c)

**Reply by PricelessToolkit** [reply]
Address of the MQTT Server needs to be in IP format "not DNS name"

**Comment by muradsam5555** [comment]
![IMG_1186](https://github.com/user-attachments/assets/d7a60625-ed99-4019-be73-c46c9ea1b5cf)
Please help me I have not been able to finish this device for 4 months

**Reply by PricelessToolkit** [reply]
If you need help, first you have to explain what is happening, what is not working, I am not a psychic, open a new discussion and I will try to help.

**Reply by muradsam5555** [reply]
nvs_open failed: NOT_FOUND

**Reply by PricelessToolkit** [reply]
1. The error message "nvs_open failed: NOT_FOUND" means that the non-volatile storage (NVS) system couldn’t find the specified key or namespace when trying to open it. This typically happens when you try to read a value that hasn’t been stored yet. so its normal.

2. You changed your MQTT server address? to IP?

**Reply by muradsam5555** [reply]
![IMG_1185](https://github.com/user-attachments/assets/5945efdc-f4d2-4b93-9a0c-e6f83ae840b4)

**Reply by muradsam5555** [reply]
Do you mean that nvs_open failed: NOT_FOUND  error is normal?

**Reply by muradsam5555** [reply]
![image](https://github.com/user-attachments/assets/ccd0aea8-6455-46e1-93b7-dbb37e423302)
Wi-Fi and mqtt connected

**Reply by muradsam5555** [reply]
Now problem only invalid input range, min== max

**Reply by PricelessToolkit** [reply]
1. Yes, it's normal because every time you access the configuration, it wipes all the connection credentials.
2. The node name is mandatory; the Gateway key is not.
3. Is "44.232.241.40" an MQTT public server (https://www.emqx.com/)?
4. Did you try connecting to that server with an MQTT client using the IP address and credentials? If it is behind a proxy server, you cannot connect to it using the IP address.
5. You can test whether the sensor is connecting to your Wi-Fi network by opening "CMD" and typing `ping 192.168.0.131 -t` if you're using Windows.

**Reply by PricelessToolkit** [reply]
ok so its connects to MQTT. and to WIFI!

**Reply by PricelessToolkit** [reply]
sensor is calibrated ?

**Reply by PricelessToolkit** [reply]
But even without calibration, you will see row ADC sensor data on the MQTT topic  "homeassistant/sensor/YourNodeName/madcrow"

**Reply by muradsam5555** [reply]
1. Yes, sensor is calibrated 
2. 
![IMG_1193](https://github.com/user-attachments/assets/c4a406aa-ff12-48bd-87eb-1e9344b1c848) but only if sensor connected via usb, the battery is connected but the light is not on
3. Now problem only invalid input range, min== max

**Reply by PricelessToolkit** [reply]
1. When on the battery, the light will turn on ~1sec every hour when the timer triggers. "it publishes MQTT payload and turns off" 
2. It publishing its data on MQTT? 
3. Can you see "Moisture" percentage ?

**Reply by muradsam5555** [reply]
invalid input range, min== max is normal? 
No, it is not publishing its data on MQTT. 
I should subscribe to Topic homeassistant/sensor/ right?

**Reply by PricelessToolkit** [reply]
homeassistant/sensor/YourNodeName/

**Reply by muradsam5555** [reply]
Can you explain to me why this error occurs and how to solve it?
 invalid input range, min== max

**Reply by PricelessToolkit** [reply]
If you don't want to answer my questions to understand why, the only option is "not calibrated!"

**Reply by PricelessToolkit** [reply]
https://github.com/PricelessToolkit/SOILSENS-V5W?tab=readme-ov-file#2-calibrating-the-soil-moisture-sensor
